commit 74a2c7b00060951c9b8cdee8100ab4fdbafd671a (refs/changes/57/1457/2)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2017-02-07T12:09:58+01:00 (2 years, 8 months ago)
    
    TOOLS-1681 sdcadm experimental update doesn't summarize properly when there are several agent updates

diff --git a/lib/procedures/update-agent-v1.js b/lib/procedures/update-agent-v1.js
index 40629f9..9259ef9 100644
--- a/lib/procedures/update-agent-v1.js
+++ b/lib/procedures/update-agent-v1.js
@@ -38,15 +38,16 @@ util.inherits(UpdateAgentV1, Procedure);
 
 
 UpdateAgentV1.prototype.summarize = function uaSummarize() {
-    var c0 = this.changes[0];
-    var img = c0.image;
-    var out = [sprintf('update "%s" service to image %s ',
-                c0.service.name, img.uuid),
-                common.indent(sprintf('%s@%s', img.name, img.version))];
-    if (c0.insts.length) {
-        out.push(sprintf('on %d servers', c0.insts.length));
-    }
-    return out.join('\n');
+    return this.changes.map(function (ch) {
+        var img = ch.image;
+        var out = [sprintf('update "%s" service to image %s ',
+                    ch.service.name, img.uuid),
+                    common.indent(sprintf('%s@%s', img.name, img.version))];
+        if (ch.insts.length) {
+            out.push(sprintf('on %d servers', ch.insts.length));
+        }
+        return out.join('\n');
+    }).join('\n');
 };
 
 // The minimal required CNAPI version (CNAPI-508, CNAPI-511):
